### PR TITLE
feat: add partial_dependence_proba for LogisticGAM

### LINKED ISSUE #426: Logistic Regression `partial_dependence` in probability units
I noticed that `LogisticRegression` `pygam` models nicely give PDP values with confidence intervals but only in the logit additive scale. That is the most relevant and intuitive scale for additive models of course, but in many applications probability is a better unit to present results on.

I am aware that using the inverse logit function (logistic/sigmoid/expit) on the `partial_dependence` output is wrong. I can fairly easily get to a function that computes the probability mean for the PDP from the gam model using the `predict_proba` but the confidence intervals is rather tricky. Is there some way I to get to them rather easily besides using the bootstrap?

Thanks!

### PR DESCRIPTION
**Pull request by moksha-hub**
## Summary
Adds `partial_dependence_proba` method to LogisticGAM to compute partial dependence values in probability scale (0-1) instead of logit scale.

## Changes
- Add `partial_dependence_proba(term, X, width, n_samples, meshgrid)` method to LogisticGAM
- Add `_sample_coef_from_cov(n_samples)` helper method for posterior sampling
- Returns probability-scale values with confidence intervals computed via posterior sampling
- Add comprehensive tests

## Motivation
Currently, `partial_dependence` returns values in logit (linear predictor) scale. For LogisticGAM, users often want probability scale for interpretability. While applying expit directly to logit values is incorrect for CIs, this implementation uses posterior sampling to correctly compute probability-scale partial dependence with proper confidence intervals.

## Usage
```python
from pygam import LogisticGAM

gam = LogisticGAM().fit(X, y)
pdeps, conf = gam.partial_dependence_proba(term=0, width=0.95)

# pdeps: partial dependence in probability scale (0-1)
# conf: confidence intervals with shape (n_samples, 2)
```

Closes #426

### CODE REVIEW

**Comment by moksha-hub** [comment]
@jonathan-taylor @szepeviktor @cbrummitt @frantzmiccoli   please have review.
Thanks for your time.

**Review by szepeviktor — CHANGES_REQUESTED** [review]
The new test functions are missing the required class wrapper. All other test functions in this file are wrapped in test classes (e.g., TestGAMCore, TestLinearGAMAPI). These standalone test functions should be organized into an appropriate test class to maintain consistency with the file's structure.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+- [Coding Standards](#coding-standards)
+- [Testing](#testing)
+
+## Code of Conduct
+
+This project and everyone participating in it is governed by the [pyGAM Code of Conduct](CODE_OF_CONDUCT.md). By participating, you are expected to uphold this code.
+
+## Getting Started
+
+- Join our [Discord](https://discord.gg/Rt8By5Jj) for discussions and updates
```
Comment (cbrummitt) [inline CONTRIBUTING.md:21]: this link gives me a "does not exist" page

File: pygam/pygam.py [hunk 2]
```diff
@@ -222,6 +230,41 @@ def _is_fitted(self):
...
+            from sklearn.utils import Tags
+
+            tags = (
+                super().__sklearn_tags__()
+                if hasattr(super(), "__sklearn_tags__")
+                else Tags()
+            )
+
+            # GAMs support sample weights in fit()
+            tags.estimator_type = "regressor"
```
Comment (cbrummitt) [inline pygam/pygam.py:262]: LogisticGAM seems closer to a classifier than a regressor. Can you say more why you are proposing to tag it with this estimator type?

File: pygam/pygam.py [hunk 3]
```diff
@@ -2649,6 +2743,136 @@ def score(self, X, y):
...
+
+        Examples
+        --------
+        >>> from pygam import LogisticGAM
+        >>> gam = LogisticGAM().fit(X, y)
+        >>> pdeps, conf = gam.partial_dependence_proba(term=0)
+        """
+        from scipy.special import expit
+
+        if not self._is_fitted:
```
Comment (cbrummitt) [inline pygam/pygam.py:2797]: `partial_dependence` has a lot more validation checks than `partial_dependence_proba`. Can you please close that gap?

File: pygam/tests/test_GAM_methods.py [hunk 4]
```diff
@@ -655,3 +655,112 @@ def test_loglikelihood(self):
...
+def test_partial_dependence_proba_width_parameter(default_X_y):
+    """
+    Test that width parameter affects confidence interval width.
+    """
+    X, y = default_X_y
+
+    gam = LogisticGAM().fit(X, y)
+
+    pdeps_95, conf_95 = gam.partial_dependence_proba(term=0, width=0.95, n_samples=50)
+    pdeps_50, conf_50 = gam.partial_dependence_proba(term=0, width=0.50, n_samples=50)
```
Comment (cbrummitt) [inline pygam/tests/test_GAM_methods.py:750]: these tests could fail in rare cases due to randomness. can you please seed the random number generator or use mocking to avoid flakiness?

File: CODE_OF_CONDUCT.md
Comment (cbrummitt) [inline CODE_OF_CONDUCT.md:1]: it'd be nice to make this a separate PR, but i gave it a quick review and it looks good

File: pygam/tests/test_GAM_methods.py (hunk 4 above)
Comment (dswah) [reply pygam/tests/test_GAM_methods.py:750]: nice to see you!

File: pygam/pygam.py [hunk 5]
```diff
@@ -2649,6 +2743,136 @@ def score(self, X, y):
         """
         return self.accuracy(X, y, None)
 
+    def partial_dependence_proba(
+        self,
+        term,
+        X=None,
+        width=0.95,
```
Comment (marcoBmota8) [inline pygam/pygam.py:2750]: For consistency with the original `partial_dependence` method in log-odds and higher analysis flexibility, keep quantiles as the argument instead of CI width. See further comments below.

Note that a width of w can be accomplished by infinitely many CIs but only one symmetric one.
